clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -538,41 +523,6 @@ private Node awaitNodeReady(FindCoordinatorRequest.CoordinatorType coordinatorTy
         return null;
     }
 
-    private void maybeWaitForProducerId() {
-        while (!forceClose && !transactionManager.hasProducerId() && !transactionManager.hasError()) {
-            Node node = null;
-            try {
-                node = awaitNodeReady(null);
-                if (node != null) {
-                    ClientResponse response = sendAndAwaitInitProducerIdRequest(node);
```

Comment:
The only other caller of `InitProducerIdRequest.Builder` is in TransactionManager#initializeTransactions which is only called in producer#iniTxns. For idempotent producer how that initPid request would be sent? Did I miss anything?

Reply:
We also build an `InitProducerIdRequest` inside `TransactionManager.resetProducerIdIfNeeded`.